src/pepr/policies/networking.ts

```diff
@@ -84,13 +104,26 @@ When(a.Service)
     if (isExempt(request, Policy.RestrictExternalNames)) {
       return request.Approve();
     }
-    if (request.Raw.spec?.type === "ExternalName") {
+
+    const isValid = checkNotExternalNameService(request.Raw.spec);
+
+    if (isValid) {
+      return request.Approve();
+    } else {
       return request.Deny("ExternalName services are not allowed.");
     }
-
-    return request.Approve();
   });
 
+/**
+ * Checks if a service is using ExternalName type
+ */
+export function checkNotExternalNameService(serviceSpec: V1ServiceSpec | undefined): boolean {
+  if (!serviceSpec) {
+    return true; // No spec means no ExternalName type
+  }
+  return serviceSpec.type !== "ExternalName";
```

Comment:
nit: slight reduction in lines of code (aligns with what we did before)
```suggestion
  return serviceSpec?.type !== "ExternalName";
```

Reply:
updated